runtime/src/accounts_db.rs

```diff
@@ -1565,21 +1565,8 @@ impl AccountsDb {
                 );
                 true
             } else {
-                let mut no_delete = false;
-                for (_slot, account_info) in account_infos {
-                    debug!(
-                        "calc_delete_dependencies()
-                        storage id: {},
-                        count len: {}",
-                        account_info.store_id,
-                        store_counts.get(&account_info.store_id).unwrap().0,
-                    );
-                    if store_counts.get(&account_info.store_id).unwrap().0 != 0 {
-                        no_delete = true;
-                        break;
-                    }
-                }
-                no_delete
+                assert_eq!(account_infos.len(), 1);
```

Comment:
Why account_infos.len(). would be = 1? 
I thought the issue could be the pub key for some 0 lamports are not returned at all in construct_candidate_clean_keys and hence we will not catch them either.